src/operator/subgraph/subgraph_property.h

```diff
@@ -92,6 +92,22 @@ class SubgraphProperty {
   // execute the operators in the subgraph.
   virtual nnvm::NodePtr CreateSubgraphNode(const nnvm::Symbol &s,
                                            const int subgraph_id = 0) const = 0;
+  // Connect subgraph internal output with external output entries. By default,
+  // each output entry will connect to an unique internal output.
+  virtual void ConnectSubgraphOutputs(
+      const nnvm::NodePtr n,
+      std::vector<nnvm::NodeEntry *> *output_entries) const {
```

Comment:
i think it's better check if `n` has the same number of outputs as `output_entries`.

Reply:
This can't be guaranteed as output_entries may have duplicated entries from n.